vuu-ui/packages/vuu-layout/src/layout-persistence/LocalLayoutPersistenceManager.ts

```diff
@@ -4,47 +4,39 @@ import { LayoutJSON, LayoutPersistenceManager } from "@finos/vuu-layout";
 import { getLocalEntity, saveLocalEntity } from "@finos/vuu-filters";
 import { getUniqueId } from "@finos/vuu-utils";
 
+const metadataSaveLocation = "layouts/metadata";
+const layoutsSaveLocation = "layouts/layouts";
+
 export class LocalLayoutPersistenceManager implements LayoutPersistenceManager {
-  saveLayout(inputLayouts: Layout[]): string {
-    const layouts = Array.from(inputLayouts);
+  saveLayout(metadata: Omit<LayoutMetadata, "id">, layout: LayoutJSON): string {
+    console.log(`Saving layout as ${metadata.name} to group ${metadata.group}...`);
+
+    const existingLayouts = this.loadLayouts();
+    const existingMetadata = this.loadMetadata();
 
-    if (layouts.length === 0) {
-      return "";
-    }
-    
-    const layout = this.deepCopy(layouts[layouts.length - 1]);
-    console.log(`Saving layout as ${layout.metadata.name} to group ${layout.metadata.group}...`);
     const id = getUniqueId();
-    layout.metadata.id = id;
-    layouts[layouts.length - 1] = layout;
-    saveLocalEntity<Layout[]>("layouts", layouts);
-    return id;
-  }
 
-  private deepCopy<T>(object: T): T {
-    return JSON.parse(JSON.stringify(object));
-  };
+    this.appendAndPersist(id, metadata, layout, existingLayouts, existingMetadata);
 
-  updateLayout(id: string, newMetadata: Omit<LayoutMetadata, "id">, newLayoutJson: LayoutJSON): void {
-    const layouts = this.getExistingLayouts();
-    const layoutJson = getLocalEntity<LayoutJSON>("api/vui");
+    return id;
+  }
 
-    if (layoutJson) {
-      layouts.filter(layout => layout.metadata.id !== id)
-      const newLayout = {json: newLayoutJson, metadata: newMetadata} as Layout;
-      layouts.push(newLayout);
-    }
+  updateLayout(id: string, metadata: Omit<LayoutMetadata, "id">, newLayoutJson: LayoutJSON): void {
+    const existingLayouts = this.loadLayouts().filter(layout => layout.id !== id);
+    const existingMetadata = this.loadMetadata().filter(metadata => metadata.id !== id);
```

Comment:
This is probably the question of whether or not to nest metadata within the layout type again. I think we can do so on the backend in a more straightforward manner (using a more object-oriented approach to generating IDs), then we can probably have a single call in the repository layer that will interact with two DB tables. That said, all this may well be possible on the frontend with a bit of redesign, but I think that would be a refactor ticket down the line if we think it's worth doing.

Reply:
Well put. Agreed!